contents/tutorials/python-analytics.md

```diff
@@ -0,0 +1,270 @@
+---
+title: How to set up analytics in Python and Flask
+date: 2024-02-22
+author: ["lior-neu-ner"]
+tags: ['product analytics']
+---
+
+import { ProductScreenshot } from 'components/ProductScreenshot'
+import EventsInPostHogLight from '../images/tutorials/python-analytics/events-light.png'
+import EventsInPostHogDark from '../images/tutorials/python-analytics/events-dark.png'
+import InsightLight from '../images/tutorials/python-analytics/insight-light.png'
+import InsightDark from '../images/tutorials/python-analytics/insight-dark.png'
+
+[Product analytics](/product-analytics) enable you to gather and analyze data about how users interact with your Python app. To show you how to set up analytics, in this tutorial we create a basic Python app with [Flask](https://flask.palletsprojects.com/), add PostHog, and use it to [capture events](/docs/product-analytics/capture-events) and [create insights](/docs/product-analytics/insights).
+
+## 1. Create a basic Flask app
+
+We start by creating a simple Flask app that has two pages:
+
+1. A `login` page where a user can enter their name, email, and company name in a form.
+2. A `dashboard` page that has some text and a button.
+
+First, make sure Python and Flask are installed. If Flask is not installed, you can install it via pip:
+
+```bash
+pip3 install Flask
+```
+
+Create a new folder for your project called `flask-analytics` and set up a basic project structure:
+
+```bash
+mkdir flask-analytics
+cd flask-analytics
+mkdir templates static
+touch app.py
+cd ./templates
+touch login.html
+touch dashboard.html
+cd ..
+```
+
+Then, we set up our app routes in `app.py`:
+
+```python file=app.y
+from flask import Flask, render_template, request, redirect, session, url_for
+
+app = Flask(__name__)
+app.secret_key = 'a_super_secret_key!' # you don't need to replace this
+
+@app.route('/')
+def index():
+    return render_template('login.html')
+
+@app.route('/login', methods=['POST'])
+def login():
+    session['name'] = request.form['name']
+    session['email'] = request.form['email']
+    session['company_name'] = request.form['company_name']
+    return redirect(url_for('dashboard'))
+
+@app.route('/dashboard')
+def dashboard():
+    if 'name' in session:
+        return render_template('dashboard.html', session=session)
+    return redirect(url_for('index'))
+
+@app.route('/api/dashboard', methods=['POST'])
+def api_dashboard():
+    # Placeholder for API logic. We'll add this later
+    return '', 204
+
+if __name__ == '__main__':
+    app.run(debug=True)
```

Comment:
Apparently Mac uses port 5000 for its [control centre](https://stackoverflow.com/questions/72795799/how-to-solve-403-error-with-flask-in-python), so adjusting this to make it work
```suggestion
    app.run(port=8000, debug=True)
```